src/main/java/com/openlattice/edm/PostgresEdmManager.java

```diff
@@ -164,21 +181,29 @@ public void revoke(
 
     }
 
-    public Iterable<EntitySet> getAllEntitySets() {
-        String getAllEntitySets = String.format( "SELECT * FROM %1$s", ENTITY_SETS );
-        try ( Connection connection = hds.getConnection();
-                PreparedStatement ps = connection.prepareStatement( getAllEntitySets );
-                ResultSet rs = ps.executeQuery() ) {
-            List<EntitySet> result = Lists.newArrayList();
-            while ( rs.next() ) {
-                result.add( ResultSetAdapters.entitySet( rs ) );
-            }
-
-            return result;
-        } catch ( SQLException e ) {
-            logger.error( "Unable to load all entity sets", e );
-            return ImmutableList.of();
-        }
+    public PostgresIterable<EntitySet> getAllEntitySets() {
+        return new PostgresIterable<>(
+                () -> {
+                    try {
```

Comment:
Prefer 
```
try ( Connection connection = hds.getConnection();
                        PreparedStatement ps = connection.prepareStatement( getAllEntitySets ) ) {
```

Reply:
This will break the postgres iterable. I hope you haven't done this anywhere :O

Reply:
😅